### PR TITLE
feat: use SPDX license headers

### PR DESCRIPTION
**Pull request by onerandomusername**
## Summary

clean up the licensing in each file by using SPDX license headers and add the headers to missing files, (nearly every file should have been edited)

*will add a bit about this to the contributing documentation*

## Checklist

- [ ] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [ ] I have formatted the code properly by running `task lint`
    - [ ] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/ext/commands/_types.py [hunk 1]
```diff
@@ -1,27 +1,4 @@
...
-
-THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
-OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
-FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
-DEALINGS IN THE SOFTWARE.
-"""
+# SPDX-License-Identifier: MIT
```
[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/ext/commands/_types.py:2]: https://github.com/DisnakeDev/disnake/pull/756/commits/d9460aecb1e6c4e39b52a7fed4dc28622c7bfbc6

File: disnake/ext/commands/base_core.py [hunk 2]
```diff
@@ -1,24 +1,5 @@
...
-
-# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
-# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
-# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
-# DEALINGS IN THE SOFTWARE.
+# SPDX-License-Identifier: MIT
+
```
[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/ext/commands/base_core.py:2]: https://github.com/DisnakeDev/disnake/pull/756/commits/d9460aecb1e6c4e39b52a7fed4dc28622c7bfbc6

File: disnake/ext/commands/interaction_bot_base.py [hunk 3]
```diff
@@ -1,24 +1,5 @@
...
-
-# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
-# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
-# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
-# DEALINGS IN THE SOFTWARE.
+# SPDX-License-Identifier: MIT
+
```
[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/ext/commands/interaction_bot_base.py:2]: https://github.com/DisnakeDev/disnake/pull/756/commits/d9460aecb1e6c4e39b52a7fed4dc28622c7bfbc6

File: disnake/interactions/application_command.py [hunk 4]
```diff
@@ -1,24 +1,5 @@
...
-
-# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
-# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
-# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
-# DEALINGS IN THE SOFTWARE.
+# SPDX-License-Identifier: MIT
+
```
[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/interactions/application_command.py:2]: https://github.com/DisnakeDev/disnake/pull/756/commits/d9460aecb1e6c4e39b52a7fed4dc28622c7bfbc6

File: disnake/interactions/base.py [hunk 5]
```diff
@@ -1,29 +1,6 @@
 # -*- coding: utf-8 -*-
```
[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/interactions/base.py]: https://github.com/DisnakeDev/disnake/pull/756/commits/d9460aecb1e6c4e39b52a7fed4dc28622c7bfbc6

File: disnake/interactions/message.py [hunk 6]
```diff
@@ -1,24 +1,5 @@
...
-
-# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
-# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
-# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
-# DEALINGS IN THE SOFTWARE.
+# SPDX-License-Identifier: MIT
+
```
[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/interactions/message.py:2]: https://github.com/DisnakeDev/disnake/pull/756/commits/d9460aecb1e6c4e39b52a7fed4dc28622c7bfbc6

File: disnake/interactions/base.py [hunk 7]
```diff
@@ -1,29 +1,6 @@
...
+# SPDX-License-Identifier: MIT
```
[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/interactions/base.py:2]: https://github.com/DisnakeDev/disnake/pull/756/commits/d9460aecb1e6c4e39b52a7fed4dc28622c7bfbc6

**Review by EQUENOS — APPROVED** [review]
Lgtm! Using SPDX license identifiers is a good practice

**Comment by onerandomusername** [comment]
The two files that have been edited a bit more than just license heading (and therefore need extra scrutiny) should be `.pre-commit-config.yaml` and `CONTRIBUTING.md`

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -51,6 +53,12 @@ task setup_env
 The installed `pre-commit` hooks will automatically run before every commit, which will format/lint the code
 to match the project's style. Note that you will have to stage and commit again if anything was updated!
 
+Most of the time, running pre-commit will automatically fix any issues that arise, but this is not always the case.
+We have a few hooks that *don't* resolve their issues automatically, and must be fixed manually.
+One of these is our license headers, which must exist in all files unless comments are not supported in those files, or they
+are not text files, in which case exceptions can be made. These headers must exist following the format
+documented at [https://spdx.dev/ids/](https://spdx.dev/ids/).
```
Comment (shiftinv) [inline CONTRIBUTING.md:60]:
```suggestion
One of these is the license header, which must exist in all files unless comments are not supported in those files, or they
are not text files, in which case exceptions can be made. These headers must exist following the format
documented at [https://spdx.dev/ids/](https://spdx.dev/ids/).
```

File: .pre-commit-config.yaml [hunk 9]
```diff
@@ -25,6 +27,15 @@ repos:
...
         exclude: '/(\d+\.[a-z]+(\.\d+)?\.rst|_template.rst.jinja|README.rst)$'
+      - id: spdx-licensing
+        name: SPDX License Identifiers
+        description: Check for the SPDX-License-Identifier in each file.
+        language: pygrep
+        entry: 'SPDX-License-Identifier: '
+        args: [--negate]
+        types: [text]
+        exclude_types: [json, pofile]
+        exclude: 'changelog/|py.typed|disnake/bin/COPYING|.github/PULL_REQUEST_TEMPLATE.md|LICENSE|MANIFEST.in'
```
Comment (shiftinv) [inline .pre-commit-config.yaml:38]:
```suggestion
        exclude: 'changelog/|py.typed|disnake/bin/COPYING|.github/PULL_REQUEST_TEMPLATE.md|LICENSE|MANIFEST.in|requirements/|requirements.txt'
```

Comment (onerandomusername) [reply .pre-commit-config.yaml:38]: why are those being excluded

Comment (shiftinv) [reply .pre-commit-config.yaml:38]: I don't see a point in adding license identifiers to requirement files, we don't need to overdo it. The `LICENSE` file always applies, regardless of the identifier in every file.

Comment (onerandomusername) [reply .pre-commit-config.yaml]: I guess? Kinda goes against the purposes of SPDX but okay
https://github.com/DisnakeDev/disnake/pull/756/commits/8faa69d1336e1132ea383250b10b3be9e6c82f02

**Review by shiftinv — APPROVED** [review]
woo

**Pull request merged**